File: fe-admin/src/app/core/model/request/search-staff-request.ts
import {PaginationFormat} from "./pagination-format";

export interface SearchStaffRequest extends PaginationFormat {
    sortBy: string;
    codeKeyword: string;
    nameKeyword: string;
    phoneKeyword: string;
    staffType: string | null;
    isActive: boolean | null;
}

File: be-api/src/main/java/cd/beapi/dto/request/SearchStaffRequest.java
package cd.beapi.dto.request;

import cd.beapi.enumerate.StaffSortOption;
import cd.beapi.enumerate.StaffType;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.experimental.FieldDefaults;

@Getter
@FieldDefaults(level = AccessLevel.PRIVATE)
public class SearchStaffRequest extends PaginationFormat {
    String codeKeyword;
    String nameKeyword;
    String phoneKeyword;
    StaffType staffType;
    Boolean isActive;
    StaffSortOption sortBy;
}
